miden-tx/src/executor/mod.rs

```diff
@@ -80,6 +81,7 @@ impl TransactionExecutor {
                 Some(self.exec_options.max_cycles()),
                 self.exec_options.expected_cycles(),
                 self.exec_options.enable_tracing(),
+                in_debug_mode,
```

Comment:
I think we may be able to simplify this as the comment on lines 77 - 79 above is no longer correct.